sbin/build.sh

```diff
@@ -253,6 +253,134 @@ patchFreetypeWindows() {
+    elif [ "${fileVersionArray[2]}" -gt 0 ]; then
+      fileVersionString="${fileVersionString}.${fileVersionArray[1]}.${fileVersionArray[2]}"
+    elif [ "${fileVersionArray[1]}" -gt 0 ]; then
+      fileVersionString="${fileVersionString}.${fileVersionArray[1]}"
+    fi
+  else
+    # jdk8 uses this format: jdk8u482-b01
+    fileVersionString="jdk8u${fileVersionArray[1]}"
+  fi
+
+  # Check to ensure that we've produced a valid version string.
+  if [[ ! "${fileVersionString}" =~ ^jdk8u[0-9]+$ ]]; then
+    if [[ ! "${fileVersionString}" =~ ^jdk\-[0-9]+[0-9\.]*$ ]]; then
+      echo "ERROR: build.sh: ${funcName}: The version string parsed from the version numbers file does not appear to use a valid format." >&2
+      exit 1
+    fi
+  fi
+
+  # The parsing is complete and the output is formatted.
+  # Returning the version string.
+  echo "${fileVersionString}"
+}
+
+# This function attempts to compare the jdk version from version-numbers.conf with arg 1.
+# We will then return whichever version is bigger/later.
+# e.g. 17.0.1+32 > 17.0.0+64
+# If any errors or unusual circumstances are detected, we simply return arg1 to avoid destabilising the build.
+# Note: For error messages, use 'echo message >&2' to ensure the error isn't intercepted by the subshell.
+compareToOpenJDKFileVersion() {
+  funcName="compareToOpenJDKFileVersion"
+  # First, sanity checking on the arg.
+  if [ $# -eq 0 ]; then
+    echo "compareToOpenJDKFileVersion_was_called_with_no_args"
+    exit 1
+  elif [ $# -gt 1 ]; then
+    echo "ERROR: build.sh: ${funcName}: Too many arguments (>1) were passed to this function." >&2
+    echo "$1"
+    exit 1
+  fi
+
+  # This verifies that arg1 was in an expected format for a jdk version string.
+  # This provides some assurance that we can parse that string in this function.
+  # JDK11 and above uses this format: jdk-21.0.10+2
+  # jdk8 uses this format: jdk8u482-b01
+  if [[ ! "$1" =~ ^jdk\-[0-9]+[0-9\.]*(\+[0-9]+)?$ ]]; then
+    if [[ ! "$1" =~ ^jdk8u[0-9]+(\-b[0-9][0-9]+)?$ ]]; then
+      echo "ERROR: build.sh: ${funcName}: The JDK version passed to this function did not match the expected format." >&2
+      echo "$1"
+      exit 1
+    fi
+  fi
+
+  # Retrieve the jdk version from version-numbers.conf (minus the build number).
+  if ! fileVersionString="$(versionNumbersFileParser)"; then
+    # This is to catch versionNumbersFileParser failures.
+    echo "ERROR: build.sh: ${funcName}: versionNumbersFileParser has failed." >&2 
+    echo "$1"
+    exit 1
+  fi
+
+  if [[ "$1" =~ ^${fileVersionString}.*$ ]]; then
+    # The file version matches the function argument.
+    echo "$1"
+    return
+  else
+    # The file version does not match the function argument.
+    # Returning the file version.
+    if [ "${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}" -eq 8 ]; then
+      fileVersionString+="-b00"
+    else
+      fileVersionString+="+0"
+    fi
+    echo "${fileVersionString}"
```

Comment:
We should issue a "WARNING" here that discovered tag does not match OpenJDK src version etc...

Reply:
Done.